devchat.go

```diff
@@ -437,12 +438,52 @@ func (u *user) pickUsername(possibleName string) {
 		possibleName = cleanName(possibleName)
 	}
 	u.name = possibleName
-	u.changeColor(*colorArr[rand.Intn(len(colorArr))]) // also sets prompt
+	var colorIndex = rand.Intn(len(colorArr))
+	u.changeColor(colorNames[colorIndex], *colorArr[colorIndex]) // also sets prompt
 }
 
-func (u *user) changeColor(color color.Color) {
+func (u *user) changeColor(colorName string, color color.Color) {
 	u.name = color.Sprint(stripansi.Strip(u.name))
-	u.color = color
+	u.color = colorName
+	u.term.SetPrompt(u.name + ": ")
+	allUsersMutex.Lock()
+	allUsers[u.id] = u.name
+	allUsersMutex.Unlock()
+	saveBansAndUsers()
+}
+
+func (u *user) getColor() *color.Color {
+	for i := 0; i < len(colorNames); i++ {
+		if colorNames[i] == u.color {
+			return colorArr[i]
+		}
+	}
+	return colorArr[0]
+}
+
+func (u *user) rainbowColor() {
+	var rainbow = [...]color.Color{
+		*green,
+		*cyan,
+		*blue,
+		*red,
+		*magenta,
+		*yellow,
+		*white,
+	}
+
+	var stripped = stripansi.Strip(u.name)
+	var buf = ""
+	colorOffset := rand.Intn(len(rainbow) - 1)
+
+	for i, s := range stripped {
+		colorIndex := (colorOffset + i) % len(rainbow)
+		buf += rainbow[colorIndex].Sprint(string(rune(s)))
+		println(colorIndex, string(rune(s)))
```

Comment:
Does this code compile? No such function as println IIRC

Reply:
Were you using the `Println()` function to test out your code?

Reply:
```suggestion
```
Yes, I forgot to remove my debug statement.

Reply:
Ah no problem lol